src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -95,6 +101,15 @@ private function processValue($value, $rootLevel = 0, $level = 0)
             }
             $invalidBehavior = $value->getInvalidBehavior();
 
+            if (ContainerInterface::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE === $invalidBehavior && $value instanceof TypedReference) {
+                $e = new ServiceNotFoundException($id = (string) $value, $this->currentId);
+
+                $this->container->register($id, $value->getType())
```

Comment:
what if a next processed service was also trying to reference this service in the container, with a stricter behavior ? The service is now defined and will throw only at runtime.

Reply:
this cannot happen between `ResolveInvalidReferencesPass` and `CheckReferenceValidityPass`, so that this case doesn't exist AFAIK

Reply:
I mean, next services processed by this same compiler pass (this logic is in a loop over services)

Reply:
Ok, got it. Better now?